pkg/operator/cluster/ceph/osd/pod.go

```diff
@@ -40,70 +44,152 @@ const (
 	locationEnvVarName        = "ROOK_LOCATION"
 )
 
-func (c *Cluster) makeDaemonSet(selection rookalpha.Selection, config rookalpha.Config) *extensions.DaemonSet {
-	podSpec := c.podTemplateSpec(nil, selection, c.resources, config)
-	return &extensions.DaemonSet{
+func (c *Cluster) makeJob(nodeName string, devices []rookalpha.Device,
+	selection rookalpha.Selection, resources v1.ResourceRequirements, config rookalpha.Config) *batch.Job {
+
+	podSpec := c.podTemplateSpec(devices, selection, resources, config, true /* prepare-only */, v1.RestartPolicyOnFailure)
+	podSpec.Spec.NodeSelector = map[string]string{apis.LabelHostname: nodeName}
+
+	return &batch.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:            appName,
+			Name:            fmt.Sprintf(prepareAppNameFmt, nodeName),
 			Namespace:       c.Namespace,
 			OwnerReferences: []metav1.OwnerReference{c.ownerRef},
 			Labels: map[string]string{
 				k8sutil.AppAttr:     appName,
 				k8sutil.ClusterAttr: c.Namespace,
 			},
 		},
-		Spec: extensions.DaemonSetSpec{
-			UpdateStrategy: extensions.DaemonSetUpdateStrategy{
-				Type: extensions.RollingUpdateDaemonSetStrategyType,
-			},
+		Spec: batch.JobSpec{
 			Template: podSpec,
 		},
 	}
 }
 
-func (c *Cluster) makeReplicaSet(nodeName string, devices []rookalpha.Device,
-	selection rookalpha.Selection, resources v1.ResourceRequirements, config rookalpha.Config) *extensions.ReplicaSet {
-
-	podSpec := c.podTemplateSpec(devices, selection, resources, config)
-	podSpec.Spec.NodeSelector = map[string]string{apis.LabelHostname: nodeName}
+func (c *Cluster) makeOSDDeployment(nodeName string, devices []rookalpha.Device, selection rookalpha.Selection, resources v1.ResourceRequirements, osd OSDInfo) *extensions.Deployment {
 	replicaCount := int32(1)
+	volumeMounts := []v1.VolumeMount{
+		{Name: k8sutil.DataDirVolume, MountPath: k8sutil.DataDir},
+		k8sutil.ConfigOverrideMount(),
+	}
+	volumes := []v1.Volume{k8sutil.ConfigOverrideVolume()}
+	if c.dataDirHostPath != "" {
```

Comment:
we will always have a dataDirHostPath now, right?

Reply:
should be the case. Added a log msg if no data path.